### PR TITLE
Add 'off' method to deregister all handlers for an event (including singles)

### PR DESCRIPTION
**Pull request by scottlazz**
This is useful for handling dynamically created elements.

Piggybacking on the work here: https://github.com/losthumans/in-view
this PR also clears the "once" listeners.

### CODE REVIEW

File: src/registry.js [hunk 1]
```diff
@@ -51,6 +51,21 @@ class inViewRegistry {
...
 
     /**
+    * Deregister all handlers for an event.
+    */
+    off(event) {
+        for (var i = this.handlers[event].length; i > 0; i--) {
+            this.handlers[event].pop();
+        }
+
+        for (var j = this.singles[event].length; j > 0; j--) {
```
Comment (camwiegert) [inline src/registry.js]: Any reason not to do it like so?

```js
this.handlers[event] = [];
this.singles[event] = [];
```

File: src/registry.js [hunk 2]
```diff
@@ -51,6 +51,21 @@ class inViewRegistry {
...
+            this.singles[event].pop();
+        }
+
+        return this;
+    }
```
Comment (camwiegert) [inline src/registry.js]: Any reason not to do this like so?

```js
this.handlers[event] = [];
this.singles[event] = [];
```

**Comment by jsonberry** [comment]
@camwiegert yessss please merge this, I'm enjoying `in-view` but I am having trouble with it in some edge cases when used in a Vue project. Handlers are getting multiplied when routing, so I have to do hacky things like wrapping methods in a setTimeout.

I definitely have use for a method like this for removing handlers.

File: src/registry.js (hunk 2 above)
Comment (scottlazz) [reply src/registry.js]: Right. That makes a lot more sense.

**Comment by tomasc** [comment]
@camwiegert would be great to have this merged!

**Comment by KasperAndersson** [comment]
What's status on this @camwiegert ? Why isn't this merged

**Pull request closed without merging**